### PR TITLE
chore: introduce Changesets for publishing

### LINKED ISSUE #1040: Improve release process
Tracking issue for work to automate release process of the devtools

### PR DESCRIPTION
**Pull request by alessbell**
Closes #1040. Introduces Changesets to automate the devtools release process.

### CODE REVIEW

**Review by jerelmiller — APPROVED** [review]
💯 💯 🎉 🎉 🔥 🔥

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+          value: ${{ steps.package-version.outputs.current-version }}
+
+      - name: Upload & release Chrome extension
+        uses: mnao305/chrome-extension-upload@v4.0.1
+        with:
+          file-path: dist/chrome.zip
+          extension-id: hogefuga(extension id)
+          client-id: ${{ secrets.CLIENT_ID }}
+          client-secret: ${{ secrets.CLIENT_SECRET }}
+          refresh-token: ${{ secrets.REFRESH_TOKEN }}
```
Comment (alessbell) [inline .github/workflows/release.yml:82]: To do: update `extension-id` and set these secrets

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+      - name: Upload & release Firefox addon
+        uses: trmcnvn/firefox-addon@v1
+        with:
+          uuid: "{7b312f5e-9680-436b-acc1-9b09f60e8aaa}"
+          xpi: "dist/apollo_client_devtools-${{ steps.package-version.outputs.current-version }}.zip"
+          manifest: src/extension/manifest.json
+          api-key: ${{ secrets.FIREFOX_API_KEY }}
+          api-secret: ${{ secrets.FIREFOX_API_SECRET }}
```
Comment (alessbell) [inline .github/workflows/release.yml:91]: To do: update `uuid` and set these two secrets

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                    {
+                      "type": "mrkdwn",
+                      "text": "*npm releases:*\n<https://www.npmjs.com/package/apollo-client-devtools/v/${{ fromJson(steps.changesets.outputs.publishedPackages)[0].version }}|link>"
+                    }
+                  ]
+                }
+              ]
+            }
+        env:
+          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
```
Comment (alessbell) [inline .github/workflows/release.yml:137]: To do: set this secret

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+      - name: Create release PR or publish to npm + GitHub
+        id: changesets
+        if: steps.check_files.outputs.files_exists == 'false'
+        uses: changesets/action@v1
+        with:
+          version: npm run changeset-version
+          publish: npm run changeset-publish
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (alessbell) [inline .github/workflows/release.yml:62]: To do: set this secret

**Comment by alessbell** [comment]
Just realized one other to do beyond just setting the necessary secrets: I'll need to get the new version number and update `src/extension/manifest.json` _before_ running the `zip` command.

File: .github/workflows/release.yml (hunk 1 above)
Comment (alessbell) [reply .github/workflows/release.yml]: It's taking a bit longer to get the perms to generate the keys for the Chrome Web Store Publish API, so I've commented it out for now to unblock this PR.

**Pull request merged**